### ISSUE TITLE
loss为nan

### ISSUE THREAD

**Issue by Sakurasilent** [lucidrains/iTransformer#35]
我简单的使用traffic数据集进行训练，
训练速度过慢（显卡利用率低，很长时间才功耗还会起来一下子）
并且loss始终为nan
训练代码如下：
`#!/usr/bin/env python
# -*- coding: utf-8 -*-
'''
@File    :   traffic_train.py
@Time    :   2025/04/17 11:07:09
@Author  :   fl
@Version :   1.0
@Desc    :   流量训练测试
'''

import torch
import torch.nn as nn
import pandas as pd 
import numpy as np 
from torch.utils.data import Dataset, DataLoader
from sklearn.preprocessing import StandardScaler
from iTransformer import iTransformer
from torch.cuda.amp import autocast, GradScaler
from torch import amp
import os
os.environ["PYTORCH_SDP_USE_FALLBACK"] = "1"

SEQ_LEN = 96
LABEL_LEN = 48  # 可不用于训练
PRED_LEN = 720
BATCH_SIZE = 32
EPOCHS = 10

LR = 1e-4
DEVICE = 'cuda' if torch.cuda.is_available() else 'cpu'
# print(DEVICE)

# 数据集定义
class TrafficDataset(Dataset):
    def __init__(self, data, seq_len, pred_len):
        self.seq_len = seq_len
        self.pred_len = pred_len
        self.data =data
        self.length = len(data) - seq_len - pred_len
    def __len__(self):
        return self.length
    def __getitem__(self, idx):
        seq_x = self.data[idx: idx + self.seq_len]
        seq_y = self.data[idx + self.seq_len:idx + self.seq_len + self.pred_len]
        return torch.tensor(seq_x, dtype = torch.float32), torch.tensor(seq_y,dtype = torch.float32)

# 数据加载

df = pd.read_csv("./data/traffic/traffic.csv")
# print(f"数据长度{len(df)}")
# print(df.head())
# print("-" * 100)
data = df.values[:,1:] # 去掉时间列
# print(data)
scalter = StandardScaler()
data = scalter.fit_transform(data)

dataset = TrafficDataset(data, SEQ_LEN, PRED_LEN)
dataloader = DataLoader(dataset, batch_size=BATCH_SIZE, shuffle=True)
# dataloader = DataLoader(dataset, batch_size=1)

print(data.shape[1])
print(data.shape)
# 构建模型
model = iTransformer(
    num_variates=data.shape[1],
    lookback_len=96,
    dim=data.shape[1],    # 862
    pred_length=PRED_LEN,
    depth=6,
    heads=8,
    ff_mult=4,
    # patch_len=24,
    # stride=24,
    # dropout=0.1
).to(DEVICE).half()

# print(model)

# 训练准备
optimizer = torch.optim.Adam(model.parameters(), lr = LR)
criterion = nn.MSELoss()

model.train()

# scaler = GradScaler()
from tqdm import tqdm
print("staring training")
for eopch in range(EPOCHS):
    total_loss = 0
    for batch_x, batch_y in tqdm(dataloader):
        # batch_x = batch_x.to(DEVICE).float()  # [B, T, D]
        # batch_y = batch_y.to(DEVICE).float()  # [B, T, D]
        # batch_x = batch_x.to(DEVICE)  # [B, T, D]
        # batch_y = batch_y.to(DEVICE)  # [B, T, D]
        batch_x = batch_x.half().to(DEVICE)  # [B, T, D]
        batch_y = batch_y.half().to(DEVICE)  # [B, T, D]
        # print(batch_x)
        # print(batch_x.shape)
        # print("-"*100)
        # print(batch_y)
        # print(batch_y.shape)
        # exit()
        optimizer.zero_grad()
        # with amp.autocast(device_type=DEVICE):  # 自动混合精度
        #     output = model(batch_x)[720]
        #     # print("-"*100)
        #     # print(output.keys())
        #     loss = criterion(output, batch_y)
        
        # scaler.scale(loss).backward()
        # scaler.step(optimizer)
        # scaler.update()
        
        
        output = model(batch_x)[720]
        loss = criterion(output, batch_y)
        loss.backward()
        optimizer.step()
        total_loss += loss.item()
        print(f"curr_loss:{loss} curr_total_loss:{total_loss}")

    print(f"[Epoch {epoch+1}/{EPOCHS}] Loss: {total_loss / len(dataloader):.4f}")
`
